src/components/molecules/MockedVisualisation.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState, useEffect } from 'react';
+import { VisualisationType } from '../../types/Metadata';
+import LineChart from '../visualisations/LineChart';
+import ThresholdChart from '../visualisations/ThresholdChart';
+import { Text } from 'evergreen-ui';
+import mockCartesianChartInput from '../../mockdata/mockCartesianChartInput';
+import { CartesianChartInput } from '../../types/ChartInput';
+
+type MockedVisualisationProps = {
+    visualisationType: VisualisationType;
+    height: string | number;
+    width: string | number;
+};
+
+const MockedVisualisation: React.FC<MockedVisualisationProps> = ({ visualisationType, height, width }) => {
+    const [data, setData] = useState<CartesianChartInput>(mockCartesianChartInput(100, visualisationType));
+
+    useEffect(() => {
+        setData(mockCartesianChartInput(100, visualisationType));
+    }, [visualisationType]);
```

Comment:
What is this for?

```ts
const data = mockCartesianChartInput(100, visualisationType);
```

Wouldn't this do the same?

Reply:
When `visualisationType` changes `data` will be recalculated anyway due to prop changes.

Try to avoid using `useEffect`s where they aren't needed. It might help to try _not_ using them before resorting to them – and you might avoid some review questions while you're at it;)

Reply:
You also don't need to specify the type of the `useState` when you're initialising the state immediately. It's inferred.

Reply:
It would create a new graph ever time the window is resized or if the visualisation is changed to a different size

Reply:
I see.

I would recommend looking into the available hooks in React, and perhaps particularly `useMemo`.

Reply:
The pattern of `useState` plus a `useEffect` is rather convoluted and difficult to understand right away, and requires duplicating code which is something we should aim to avoid.